### DISCUSSION TITLE
Practicum Submission | Saiful Rijal

### DISCUSSION THREAD

**Discussion by saifulrijal-ds** [onlyphantom/llm-python#43] [Practicum: Generative AI]
Hello,

Please find my project submission for the challenge here Google Colab Link: [Saiful_Generative AI and Chatbot_challenge.ipynb](https://colab.research.google.com/drive/1ERN9pPFsATXAaGYO0gNmOf3g07N6h90f?usp=sharing)

This has been an exciting challenge, and I hope my work is both timely and valuable. Thank you for the opportunity!

**Comment by saifulrijal-ds** [comment]
Key Takeaways from the Challenge
- Surprising Model Behavior: In Part 1, I noticed that smarter models can sometimes miss the mark. For example, while extracting company symbols, some models “know” the correct symbol and return it—even if the text itself is wrong. This shows that more intelligence doesn’t always mean more accuracy!
- Importance of System Messages: Part 2 highlighted just how essential system messages are. They don’t just format the output but also make sure that each tool gets the right input. In this challenge, setting up the system message correctly was key to keeping everything on track.

**Reply by onlyphantom** [reply]
I noticed sorry for grading this so late, there are perhaps more specific, robust ways to write 
- test_output_parser_symbol_invalid
- test_output_parser_mcap_invalid (checking if extraction of market cap is correctly above 0) 

But nevertheless good effort, grading it a pass! I need your email / sectors.app username to issue a certificate :)
